src/core/datagen/DataGen.jl

```diff
@@ -66,7 +66,7 @@ function get_payload_group_id(
 )
     percentile = (rank + 1) / nranks
     current = 0.0
-    for id = 1:length(cfg.payload_groups)
+    for id in eachindex(cfg.payload_groups)
```

Comment:
Since we don't need the index explicitly but the element should we do a `for payload_group in cfg.payload_groups`? Unless I'm mistaken in the nature of the `cfg.payload_groups` container.

Reply:
This function returns the `id` of the payload group that matches the current MPI rank.